x-pack/plugins/security_solution/public/flyout/right/components/description.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButtonEmpty, EuiFlexGroup, EuiFlexItem, EuiTitle } from '@elastic/eui';
+import type { VFC } from 'react';
+import React, { useState } from 'react';
+import { css } from '@emotion/react';
+import { useRightPanelContext } from '../context';
+import { useBasicDataFromDetailsData } from '../../../timelines/components/side_panel/event_details/helpers';
+import {
+  DESCRIPTION_DETAILS_TEST_ID,
+  DESCRIPTION_EXPAND_BUTTON_TEST_ID,
+  DESCRIPTION_TITLE_TEST_ID,
+} from './test_ids';
+import {
+  DOCUMENT_DESCRIPTION_COLLAPSE_BUTTON,
+  DOCUMENT_DESCRIPTION_EXPAND_BUTTON,
+  DOCUMENT_DESCRIPTION_TITLE,
+  RULE_DESCRIPTION_TITLE,
+} from './translations';
+
+export interface DescriptionProps {
+  /**
+   * Boolean to allow the component to be expanded or collapsed on first render
+   */
+  expanded?: boolean;
+}
+
+/**
+ * Displays the description of a document.
+ * If the document is an alert we show the rule description. If the document is of another type, we show -.
+ * By default, the text is truncated to only shows 2 lines.
+ * The Expand/Collapse button allows the user to see the whole description.
+ */
+export const Description: VFC<DescriptionProps> = ({ expanded = false }) => {
+  const [isExpanded, setIsExpanded] = useState(expanded);
+
+  const { dataFormattedForFieldBrowser } = useRightPanelContext();
+  const { isAlert, ruleDescription } = useBasicDataFromDetailsData(dataFormattedForFieldBrowser);
+
+  if (!dataFormattedForFieldBrowser) {
+    return null;
+  }
+
+  // TODO look into hiding the expand/collapse button if the description is short
+  //  see https://github.com/elastic/security-team/issues/6248
```

Comment:
👍🏾 this clarifies my earlier question, thanks

Reply:
ooooops, thanks for catching this!